### PR TITLE
feat(utxo-lib): add thirdparty fixtures

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
All bitcoin core forks (so far) have static test vectors, which include
tests for valid transactions and signature hashes.

Add script to download and extract these vectors as well as some tests to
verify those vectors.

Issue: BG-34381

057cae5a2
fix(utxo-lib): write `version` as `Int32`.

It gets parsed as Int32 so we should write it as Int32 as well.

See also https://github.com/bitcoinjs/bitcoinjs-lib/pull/1717

Issue: BG-34381

2d5f7e6bf
fix(utxo-lib): fix `addForkId` evaluation

Turns out `>` has higher precedence (12) than `&`!

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence

Issue: BG-34381

### CODE REVIEW

File: modules/utxo-lib/test/fixtures_thirdparty/parse.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        }
+
+        function runCheckHashForSignature([
+          rawTransaction,
+          script,
+          inputIndex,
+          hashType,
+          signatureHash,
+        ]: SigHashTestVector) {
+          if (isBitcoinGold(network) && hashType & Transaction.SIGHASH_FORKID) {
```
Comment (tylerlevine) [inline modules/utxo-lib/test/fixtures_thirdparty/parse.ts:29]: Nit: add parens to make precedence clear here

File: modules/utxo-lib/test/fixtures_thirdparty/parse.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          assert.strict(refSignatureHash.equals(hash));
+        }
+
+        testFixtureArray(network, sigHashTestFile, function (vectors: SigHashTestVector[]) {
+          if (isZcash(network)) {
+            return this.skip();
+          }
+          vectors.forEach((v) => {
+            try {
+              runCheckHashForSignature(v);
```
Comment (tylerlevine) [inline modules/utxo-lib/test/fixtures_thirdparty/parse.ts:53]: Why wrap in `try`/`catch` just to rethrow?

**Review by tylerlevine — DISMISSED** [review]

File: modules/utxo-lib/test/fixtures_thirdparty/parse.ts (hunk 2 above)
Comment (OttoAllmendinger) [reply modules/utxo-lib/test/fixtures_thirdparty/parse.ts:53]: oops

(there was a console.log with the vector index)

File: modules/utxo-lib/test/fixtures_thirdparty/parse.ts (hunk 1 above)
Comment (OttoAllmendinger) [reply modules/utxo-lib/test/fixtures_thirdparty/parse.ts:29]: `prettier` strips them off again :(

**Review by tylerlevine — APPROVED** [review]

**Comment by OttoAllmendinger** [comment]
lint failures for `account-lib` :(

**Pull request merged**